Framework/Core/include/Framework/ASoA.h

```diff
@@ -140,22 +148,54 @@ class ColumnIterator : ChunkingPolicy
+    }
+ 
     mCurrentChunk++;
-    auto array = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
-    mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    chunkToUse = mColumn->chunk(mCurrentChunk);
+    if constexpr (std::is_same_v<arrow_array_for_t<T>,arrow::FixedSizeListArray>) {
+      chunkToUse = std::dynamic_pointer_cast<arrow::FixedSizeListArray>(chunkToUse)->values();
+      auto array = std::static_pointer_cast<arrow_array_for_t<element_for_t<T>>>(chunkToUse);
+      mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
+      mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    } else {
+      auto array = std::static_pointer_cast<arrow_array_for_t<T>>(chunkToUse);
+      mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
+      mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    }
   }
 
   void prevChunk() const
   {
-    auto previousArray = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mFirstIndex -= previousArray->length();
+    auto chunkToUse = mColumn->chunk(mCurrentChunk);
+    if constexpr (std::is_same_v<arrow_array_for_t<T>,arrow::FixedSizeListArray>) {
+      chunkToUse = std::dynamic_pointer_cast<arrow::FixedSizeListArray>(chunkToUse)->values();
+      auto previousArray = std::static_pointer_cast<arrow_array_for_t<element_for_t<T>>>(chunkToUse);
+      mFirstIndex -= previousArray->length();
+    } else {
+      auto previousArray = std::static_pointer_cast<arrow_array_for_t<T>>(chunkToUse);
+      mFirstIndex -= previousArray->length();
+    }
+ 
     mCurrentChunk--;
-    auto array = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
-    mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    chunkToUse = mColumn->chunk(mCurrentChunk);
+    if constexpr (std::is_same_v<arrow_array_for_t<T>,arrow::FixedSizeListArray>) {
+      chunkToUse = std::dynamic_pointer_cast<arrow::FixedSizeListArray>(chunkToUse)->values();
+      auto array = std::static_pointer_cast<arrow_array_for_t<element_for_t<T>>>(chunkToUse);
+      mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
+      mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    } else {
+      auto array = std::static_pointer_cast<arrow_array_for_t<T>>(chunkToUse);
+      mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset() - (mFirstIndex >> SCALE_FACTOR);
+      mLast = mCurrent + array->length() + (mFirstIndex >> SCALE_FACTOR);
+    }
```

Comment:
How about hiding the differences behind helper functions rather than the if constexpr? One could return the chunk to use, for example, based on the input type (in the helper itself you can of course use if constexpr). I'd rather keep prevChunk clean from too many if (even if constexpr).